src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerEngine.java

```diff
@@ -83,8 +86,26 @@ public class HaplotypeCallerEngine implements AssemblyRegionEvaluator {
 
     protected final OutputStreamWriter assemblyDebugOutStream;
 
-    // the genotyping engine for the isActive() determination
-    private MinimalGenotypingEngine activeRegionEvaluationGenotyperEngine = null;
+    /**
+     * List of interval file entries including regions and custom ploidy values to apply in that region.
+     */
+    protected final List<NamedFeature> ploidyRegions = new ArrayList<>();
```

Comment:
I agree with James' suggestion below about using `OverlapDetector`.  Maybe here is where it happens.

Reply:
Nice suggestion, thanks! I switched to using this instead